x-pack/plugins/licensing/server/licensing_plugin_setup.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { createHash } from 'crypto';
+import { LicenseFeature } from './license_feature';
+import { LICENSE_STATUS, LICENSE_TYPE } from './constants';
+import { LicenseType, ILicensingPluginSetup } from './types';
+
+function toLicenseType(minimumLicenseRequired: LICENSE_TYPE | string) {
+  if (typeof minimumLicenseRequired !== 'string') {
+    return minimumLicenseRequired;
+  }
+
+  if (!(minimumLicenseRequired in LICENSE_TYPE)) {
+    throw new Error(`${minimumLicenseRequired} is not a valid license type`);
+  }
+
+  return LICENSE_TYPE[minimumLicenseRequired as LicenseType];
+}
+
+export class LicensingPluginSetup implements ILicensingPluginSetup {
+  private readonly hasLicense: boolean;
+  private readonly license: any;
+  private readonly features: any;
+  private _signature!: string;
+  private objectified!: any;
+  private featuresMap: Map<string, LicenseFeature>;
```

Comment:
nit: You could mark this field as readonly since the Map instance itself should never change.